src/simsopt/mhd/vmec.py

```diff
@@ -523,7 +528,9 @@ def set_indata(self):
         mpol_capped = np.min([boundary_RZFourier.mpol, 101])
         ntor_capped = np.min([boundary_RZFourier.ntor, 101])
         # Transfer boundary shape data from the surface object to VMEC:
-        for m in range(mpol_capped + 1):
+        # The highest mode number that VMEC can use is `m=100`
+        # if the max resolution is `mpol=101`.
+        for m in range(mpol_capped):
```

Comment:
It doesn't seem correct to remove the +1 here. mpol_capped is essentially mpol of a SurfaceRZFourier object (via `boundary_RZFourier = self.boundary.to_RZFourier()` and `mpol_capped = np.min([boundary_RZFourier.mpol, 101])`. So the m=mpol mode should be included rather than excluded. If you want to change the 101 to 100 on line 523/528, that would be fine.